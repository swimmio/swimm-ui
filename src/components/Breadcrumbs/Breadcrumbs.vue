<template>
  <div class='breadcrumbs'>
    <Breadcrumb v-for='(item, index) in items'
                :key='index'
                class='item'
                :name='item.name'
                :icon='item.icon'
                :link='item.link'
    />
  </div>
</template>

<script>
import Breadcrumb from './Breadcrumb/Breadcrumb.vue';

export default {
  components: { Breadcrumb },
  props: {
    items: {
      type: Array,
      required: true,
      validator: (items) => {
        return items.every(({ icon, name }) => {
          return icon && name;
        });
      }
    }
  }
};
</script>
<style scoped>
.breadcrumbs {
  display: flex;
  flex-wrap: wrap;
}

.item:not(:last-child):after {
  content: '/';
  padding: 10px;
}
</style>
