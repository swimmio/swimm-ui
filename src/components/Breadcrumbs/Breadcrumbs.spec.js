import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Breadcrumbs from './Breadcrumbs.vue';

describe('Breadcrumbs', () => {
  it('renders empty array', async () => {
    const wrapper = shallowMount(Breadcrumbs, {
      propsData: { items: [] },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders default Breadcrumbs', async () => {
    const wrapper = shallowMount(Breadcrumbs, {
      propsData: {
        items: [{ icon: 'g', name: 'g' }],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders breadcrumbs with links', async () => {
    const wrapper = shallowMount(Breadcrumbs, {
      propsData: {
        items: [
          { icon: 'g', name: 'g', link: '/g' },
          { icon: 'b', name: 'b', link: '/b' },
        ],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
