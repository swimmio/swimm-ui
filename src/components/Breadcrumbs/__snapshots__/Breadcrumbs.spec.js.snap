// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders breadcrumbs with links 1`] = `
<div class="breadcrumbs">
  <breadcrumb-stub icon="g" name="g" link="/g" class="item"></breadcrumb-stub>
  <breadcrumb-stub icon="b" name="b" link="/b" class="item"></breadcrumb-stub>
</div>
`;

exports[`Breadcrumbs renders default Breadcrumbs 1`] = `
<div class="breadcrumbs">
  <breadcrumb-stub icon="g" name="g" class="item"></breadcrumb-stub>
</div>
`;

exports[`Breadcrumbs renders empty array 1`] = `<div class="breadcrumbs"></div>`;
